src/main/java/org/gradle/profiler/ScenarioLoader.java

```diff
@@ -212,7 +212,6 @@ public ScenarioLoader(GradleBuildConfigurationReader gradleBuildConfigurationRea
             String title = scenario.hasPath(TITLE) ? scenario.getString(TITLE) : null;
 
             List<BuildMutator> mutators = BUILD_MUTATOR_CONFIGURATORS.entrySet().stream()
-                .filter(entry -> scenario.hasPath(entry.getKey()))
```

Comment:
BuildMutators are responsible for returning BuildMutator.NOOP if they don't support this key during configuration. This centralizes the logic to the BuildMutator instead of in 2 places.